core/opendaq/signal/include/opendaq/binary_data_packet_impl.h

```diff
@@ -188,6 +184,31 @@ ErrCode BinaryDataPacketImpl<ExternalMemory>::getRawDataSize(SizeT* rawDataSize)
     return OPENDAQ_SUCCESS;
 }
 
+template <bool ExternalMemory>
+inline ErrCode INTERFACE_FUNC BinaryDataPacketImpl<ExternalMemory>::getLastValue(IBaseObject** value)
+{
+    OPENDAQ_PARAM_NOT_NULL(value);
+
+    if (dataDescriptor.getDimensions().getCount() != 0 || dataDescriptor.getSampleType() != SampleType::Binary)
+        return OPENDAQ_IGNORED;
+
+    {
+        auto descriptorStructFields = dataDescriptor.getStructFields();
+        if (descriptorStructFields.assigned() && !descriptorStructFields.empty())
+            return OPENDAQ_IGNORED;
+    }
+
+    void* addr;
+    ErrCode err = this->getData(&addr);
+    if (OPENDAQ_FAILED(err))
+        return err;
+
+    auto data = static_cast<bool*>(addr);
+    *value = Boolean(data[0]).detach();
+
+    return OPENDAQ_SUCCESS;
+}
```

Comment:
BinaryData packets are a special kind of packet where they always contain exactly 1 sample and the size of the data is provided in SampleSize (in bytes). Readers then need to know how exactly to interpret the bytes. So - it's not Boolean type data :)